client/tests/integration/upgrade.rs

```diff
@@ -380,6 +364,46 @@ fn migration_should_cause_upgrade_event() {
     assert!(!data_model.permissions.is_empty());
 }
 
+#[test]
+fn define_custom_parameter() {
+    let (rt, _peer, client) = <PeerBuilder>::new().with_port(10_996).start_with_runtime();
+    wait_for_genesis_committed(&vec![client.clone()], 0);
+
+    let (sender, mut receiver) = mpsc::channel(1);
+    let events_client = client.clone();
+
+    let _handle = rt.spawn(async move {
+        let mut stream = events_client
+            .listen_for_events_async([ExecutorEventFilter::new()])
+            .await
+            .unwrap();
+        while let Some(event) = stream.try_next().await.unwrap() {
+            if let EventBox::Data(DataEvent::Executor(ExecutorEvent::Upgraded(ExecutorUpgrade {
+                new_data_model,
+            }))) = event
+            {
+                let _ = sender.send(new_data_model).await;
+            }
+        }
+    });
+
+    upgrade_executor(
+        &client,
+        "tests/integration/smartcontracts/executor_with_custom_parameter",
+    )
+    .unwrap();
+
+    let data_model = rt
+        .block_on(async {
+            tokio::time::timeout(std::time::Duration::from_secs(60), receiver.recv()).await
```

Comment:
Imo there is no need to separate channel, you can inline consuming events form the stream here.

Reply:
can you check now?

Reply:
fixed now?

Reply:
yes